### PR TITLE
yamet pure CPU rewrite

### PR DESCRIPTION
**Pull request by atchox**
This is a pure CPU based approach to yamet. This rewrite stems from the linear time sample entropy update. For the computation of sample entropies, we only keep track of the counts of the different template types. We can do this at the time of cell file parsing with a window based approach.

### Changes

1. No need for a GPU
2. Simpler code with fewer steps
3. Extremely memory efficient as we don't store entire cell files anymore
4. `cmake` used for multi platform compatibility
5. version control added to CMake
6. Per region Shannon entropy added. Can be accessed via `--det-out`.
7. Per region and per file average methylation added. Can be accessed via `--det-out` and `--out` respectively.

### Arguments removed

1. `--n-streams` - not using CUDA anymore
2. `--print-tsv` - cannot print parsed cell files anymore (at least not easily) as we do not store entire files

### CODE REVIEW

File: method/build.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Set the build directory (relative to the current working directory)
+BUILD_DIR="build"
+
+# Create the build directory if it doesn't exist
+if [ ! -d "$BUILD_DIR" ]; then
+  mkdir "$BUILD_DIR"
+fi
+
+# Run CMake from the parent directory, pointing to the build directory
+cmake -DCMAKE_INSTALL_PREFIX="$INSTALL_PREFIX" -B"$BUILD_DIR" -S"$PWD"
```
Comment (imallona) [inline method/build.sh:15]: Does it depend on specific Boost libraries versions? I'm trying with 1.74 and letting cmake find them auto(magi)cally

Comment (imallona) [reply method/build.sh:15]: I'd suggest adding a cmake check to ensure gcc>13 @atchox

File: .github/workflows/memleaks.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+name: memleaks
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+env:
+  PREFIX: $HOME/build/
```
Comment (imallona) [inline .github/workflows/memleaks.yml:8]: This is no longer valid due to changes to the cmake call / build.sh recipe

File: align_old.cpp
Comment (imallona) [inline align_old.cpp:1]: If not sourced, why keeping it? 

If it's for archiving purposes we could tag the last version and release it as v0.01, documented with its old capabilities.

Related: should we decide on a semantic versioning release pattern, and start adding changelogs?

File: method/build.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# Set the install prefix (you can modify this to another path if needed)
+INSTALL_PREFIX="$HOME/.local"
```
Comment (imallona) [inline method/build.sh:4]: This path is not what `.github/workflows/memleaks.yml` expects, time to update the `memleaks.yml` action?

File: method/include/boost.h [hunk 4]
```diff
@@ -11,12 +11,11 @@ std::vector<std::string> getTsvFiles(const po::variables_map &vm);
 std::string              getBed(const po::variables_map &vm);
```
Comment (imallona) [inline method/include/boost.h:11]: worth adding to the documentation (and `--help` flag) how do the TSVs look like, e.g. which column is `methylated reads` and which column is `total number of reads`? Worth adding a quick check `total number of reads > methylated reads` to be on the safe side?

File: align_old.cpp
Comment (atchox) [reply align_old.cpp:1]: memleaks updated and issue fixed as seen in [this run](https://github.com/imallona/yamet/actions/runs/12330971674/job/34417039757?pr=9).

File: method/build.sh (hunk 3 above)
Comment (atchox) [reply method/build.sh]: memleaks updated and issue fixed as seen in [this run](https://github.com/imallona/yamet/actions/runs/12330971674/job/34417039757?pr=9).

File: align_old.cpp
Comment (atchox) [reply align_old.cpp:1]: We will have a stable build after we merge this PR. We should start a release pattern then.

File: .github/workflows/memleaks.yml (hunk 2 above)
Comment (atchox) [reply .github/workflows/memleaks.yml:8]: discussed in newer comment

File: method/include/boost.h (hunk 4 above)
Comment (atchox) [reply method/include/boost.h]: added some more information to the help flag and the checking for the number of reads as well

**Review by imallona — APPROVED** [review]

**Pull request merged**